rigs/heft-web-rig/shared/webpack-base.config.js

```diff
@@ -0,0 +1,270 @@
+'use strict';
+
+// Import the Webpack instance installed by heft-webpack5-plugin
+const nodeCoreLibrary = require('@rushstack/node-core-library');
+const webpack = require(nodeCoreLibrary.Import.resolveModule({
+  modulePath: 'webpack',
+  baseFolderPath: nodeCoreLibrary.Import.resolvePackage({
+    packageName: '@rushstack/heft-webpack5-plugin',
+    baseFolderPath: __dirname
+  })
+}));
```

Comment:
@iclanton Have you encountered this problem?  In order to access webpack APIs, a config needs to do `require('webpack')` however we want the instance from the plugin.  Whereas if the rig has a direct dependency on`webpack`, it might be a different version or instance.

A couple other ways we might improve this:
1. Implement a helper API `Import.require({ packageName: 'webpack', relativeToPacakge: '@rushstack/heft-webpack5-plugin' })` that is syntax sugar for the above code
2. Maybe  `@rushstack/heft-webpack5-plugin` could reexport its webpack instance, and then we'd do `require('@rushstack/heft-webpack5-plugin/dependency/webpack')`

Reply:
Maybe `heft-web-rig` must have its own dependency on Webpack.  After reenabling `strictPeerDependencies` (which was apparently disabled by accident a few months go during an upgrade), I'm getting this install error:

```
node_modules/.pnpm/node-sass@6.0.1/node_modules/node-sass: Running postinstall script, done in 265ms
 WARN  Issues with peer dependencies found
../../rigs/heft-web-rig
├─┬ css-loader
│ └── ✕ missing peer webpack@^5.0.0
├─┬ css-minimizer-webpack-plugin
│ └── ✕ missing peer webpack@^5.0.0
├─┬ html-webpack-plugin
│ └── ✕ missing peer webpack@^5.20.0
├─┬ mini-css-extract-plugin
│ └── ✕ missing peer webpack@^5.0.0
├─┬ postcss-loader
│ └── ✕ missing peer webpack@^5.0.0
├─┬ sass-loader
│ └── ✕ missing peer webpack@^5.0.0
├─┬ source-map-loader
│ └── ✕ missing peer webpack@^5.0.0
├─┬ style-loader
│ └── ✕ missing peer webpack@^5.0.0
├─┬ terser-webpack-plugin
│ └── ✕ missing peer webpack@^5.1.0
└─┬ url-loader
  └── ✕ missing peer webpack@"^4.0.0 || ^5.0.0"
Peer dependencies that should be installed:
  webpack@">=5.20.0 <6.0.0"
```

Reply:
In my latest iteration, I have re-added `webpack` as a dependency of the rig.  It feels a bit janky, but I can't think of another solution besides (1) moving all these Webpack plugins into `heft-webpack5-plugin`, OR (2) changing `heft-webpack5-plugin` to have a peer dependency on `webpack` instead of a regular dependency.

Maybe it's okay as-is.

Reply:
I've thought about passing the webpack instance to the function in `webpack.config.js`, and we could pass it to the `configureWebpack` hook. Or we could pass the resolved path to the webpack package.

Reply:
@iclanton seems like that wouldn't help with the two conflicting requirements here:

1. The `heft-web-rig` package has dependencies (loaders) that have a peer dependency on the `webpack` package. That seems to force `heft-web-rig` to depend on `webpack`.
2. The `heft-webpack5-plugin` package wants to install its own version of `webpack`.

Reply:
✔ We resolved this by converting "webpack" to be a peer dependency of `heft-webpack?-plugin`.